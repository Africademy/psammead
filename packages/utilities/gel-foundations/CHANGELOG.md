# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
