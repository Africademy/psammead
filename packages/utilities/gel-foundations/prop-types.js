module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
