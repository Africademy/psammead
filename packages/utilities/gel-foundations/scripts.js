/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
