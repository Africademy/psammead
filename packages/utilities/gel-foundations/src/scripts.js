/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
