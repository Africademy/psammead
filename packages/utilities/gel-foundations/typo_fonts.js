module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
