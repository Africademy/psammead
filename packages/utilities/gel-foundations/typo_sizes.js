module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
