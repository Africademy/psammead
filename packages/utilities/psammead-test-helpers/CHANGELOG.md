# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
