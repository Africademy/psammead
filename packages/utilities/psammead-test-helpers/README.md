# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactjs.org/docs/how-to-contribute.html#your-first-pull-request)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install @bbc/psammead-test-helpers --save
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
