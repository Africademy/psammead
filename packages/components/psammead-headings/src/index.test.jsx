import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin, arabic } from '@bbc/gel-foundations/scripts';
import { Headline, SubHeading } from './index';

describe('Headline component', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <Headline script={latin}>This is my headline.</Headline>,
  );

  shouldMatchSnapshot(
    'should render correctly with arabic script typography values',
    <Headline script={arabic}>هذا هو العنوان الخاص بي</Headline>,
  );
});

describe('SubHeading component', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <SubHeading script={latin}>This is a SubHeading</SubHeading>,
  );

  shouldMatchSnapshot(
    'should render correctly with arabic script typography values',
    <SubHeading script={arabic}>هذا عنوان فرعي</SubHeading>,
  );

  shouldMatchSnapshot(
    'should render correctly with an ID',
    <SubHeading id="This-is-a-SubHeading" script={latin}>
      This is a SubHeading
    </SubHeading>,
  );
});
