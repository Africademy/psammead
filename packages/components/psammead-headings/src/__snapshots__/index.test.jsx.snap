// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component should render correctly 1`] = `
<h1
  className="Headline-sc-1kh1qhu-0 edbytY"
>
  This is my headline.
</h1>
`;

exports[`SubHeading component attribute id should render without double quotes 1`] = `
<h2
  className="SubHeading-sc-1kh1qhu-1 egmKQa"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;

exports[`SubHeading component attribute id should render without exclamation marks 1`] = `
<h2
  className="SubHeading-sc-1kh1qhu-1 egmKQa"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;

exports[`SubHeading component attribute id should render without quotes 1`] = `
<h2
  className="SubHeading-sc-1kh1qhu-1 egmKQa"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;

exports[`SubHeading component should render correctly 1`] = `
<h2
  className="SubHeading-sc-1kh1qhu-1 egmKQa"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;
